src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -26,64 +25,66 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
                 else
                 {
-                    profileObject = profilesObject[profileName] as JObject;
-
-                    if (profileObject == null)
+                    if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.Type != JsonValueType.Object)
                     {
                         return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfileIsNotAJsonObject);
                     }
                 }
 
-                if (profileObject == null)
+                if (profileObject.Type == default)
                 {
-                    foreach (var prop in profilesObject.Properties())
+                    foreach (var prop in profilesObject.EnumerateObject())
                     {
-                        var profile = prop.Value as JObject;
-
-                        if (profile != null)
+                        if (prop.Value.Type == JsonValueType.Object)
                         {
-                            var cmdName = profile[CommandNameKey]?.Value<string>();
-                            if (_providers.ContainsKey(cmdName))
+                            if (prop.Value.TryGetProperty(CommandNameKey, out var commandNameElement) && commandNameElement.Type == JsonValueType.String)
                             {
-                                profileObject = profile;
-                                break;
+                                if (_providers.ContainsKey(commandNameElement.GetString()))
+                                {
+                                    profileObject = prop.Value;
+                                    break;
+                                }
                             }
                         }
                     }
                 }
 
-                var commandName = profileObject?[CommandNameKey]?.Value<string>();
+                if (profileObject.Type == default)
+                {
+                    return new LaunchSettingsApplyResult(false, LocalizableStrings.UsableLaunchProfileCannotBeLocated);
+                }
 
-                if (profileObject == null)
+                if (!profileObject.TryGetProperty(CommandNameKey, out var finalCommandNameElement)
+                    || finalCommandNameElement.Type != JsonValueType.String)
                 {
                     return new LaunchSettingsApplyResult(false, LocalizableStrings.UsableLaunchProfileCannotBeLocated);
                 }
 
+                string commandName = finalCommandNameElement.GetString();
                 if (!TryLocateHandler(commandName, out ILaunchSettingsProvider provider))
                 {
                     return new LaunchSettingsApplyResult(false, string.Format(LocalizableStrings.LaunchProfileHandlerCannotBeLocated, commandName));
                 }
 
                 return provider.TryApplySettings(model, profileObject, ref command);
             }
-            catch (JsonException ex)
+            catch (Exception ex)
```

Comment:
This shouldn't be exception.

Reply:
Updated the exception type to cover only parse failures, to be consistent with the previous behavior